packages/core/src/index.ts

```diff
@@ -301,9 +304,20 @@ export class Abby<
 
     const defaultValue = this._cfg.settings?.flags?.defaultValue;
 
-    // return the defaultValue if exists
-    if (storedValue === undefined && defaultValue != null) {
-      return defaultValue;
+    if (storedValue === undefined) {
```

Comment:
cleaning stuff may help here..

```suggestion
    if (storedValue !== undefined) { do the normal stuff }
    
    // fallback?
    
    // default?
```